### PR TITLE
Several Gen 8 Move Effects

### PR DESCRIPTION
**Pull request by ghoulslash**
## Description
Adds:

1) Eerie Spell
2) Surging Strikes
3) Jungle Healing
jungle_healing
4) Triple Axel
5) Coaching
coaching
6) Lash out
7) Burning Jealousy
burning_jealousy
8) Grassy Glide
9) Misty Explosion
10) Steel Roller (should probably merge #1295 first)
11) Obstruct
obsruct
12) Behemoth Blade/Bash (sort of. sets up power boost with `#if B_DYNAMAX`)
13) Decorate
14) Snipe Shot

### CODE REVIEW

**Comment by AmbientDinosaur** [comment]
This pull request needs to be tweaked a bit, with regard to Surging Strikes. As it is, Surging Strikes currently gets affected by the accuracy check that decides the amount of strikes Triple Kick (and Triple Axel) will do.

The multi-strike count of Surging Strikes is not affected by accuracy, it is always set to be three.

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -1171,13 +1171,15 @@ bool32 IsBattlerProtected(u8 battlerId, u16 move)
...
     else if (gSideStatuses[GetBattlerSide(battlerId)] & SIDE_STATUS_WIDE_GUARD
              && gBattleMoves[move].target & (MOVE_TARGET_BOTH | MOVE_TARGET_FOES_AND_ALLY))
         return TRUE;
     else if (gProtectStructs[battlerId].banefulBunkered)
         return TRUE;
+    else if (gProtectStructs[battlerId].obstruct && !IS_MOVE_STATUS(move))
+        return TRUE;
     else if (gProtectStructs[battlerId].spikyShielded)
         return TRUE;
     else if (gProtectStructs[battlerId].kingsShielded && gBattleMoves[move].power != 0)
```
Comment (ultima-soul) [inline src/battle_script_commands.c:1185]: I wonder if some of these `gBattleMoves[move].power != 0` checks should be changed to `!IS_MOVE_STATUS(move)` checks since I think they're meant to represent a check for if it's a status move being used.

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -1252,7 +1254,7 @@ static bool32 TryAegiFormChange(void)
 static void Cmd_attackcanceler(void)
 {
     s32 i, moveType;
-
+
```
Comment (ultima-soul) [inline src/battle_script_commands.c:1257]:
```suggestion

```

File: include/battle.h [hunk 3]
```diff
@@ -117,6 +117,7 @@ struct ProtectStruct
     u32 spikyShielded:1;
     u32 kingsShielded:1;
     u32 banefulBunkered:1;
+    u32 obstruct:1;
```
Comment (ultima-soul) [inline include/battle.h:120]: Maybe
```suggestion
    u32 obstructed:1;
```
to match the other protecting move flag names.

File: src/battle_script_commands.c [hunk 4]
```diff
@@ -9016,11 +9137,16 @@ static u32 ChangeStatBuffs(s8 statValue, u32 statId, u32 flags, const u8 *BS_ptr
             gBattleTextBuff2[index] = STRINGID_STATFELL >> 8;
             index++;
             gBattleTextBuff2[index] = B_BUFF_EOS;
-
+
```
Comment (ultima-soul) [inline src/battle_script_commands.c:9140]:
```suggestion

```

Comment (ultima-soul) [inline src/battle_script_commands.c:9140]:
```suggestion

```

File: src/battle_util.c [hunk 5]
```diff
@@ -6686,6 +6688,20 @@ static u16 CalcMoveBasePower(u16 move, u8 battlerAtk, u8 battlerDef)
         if (gBattleMoves[gLastUsedMove].effect == EFFECT_FUSION_COMBO && move != gLastUsedMove)
             basePower *= 2;
         break;
+    case EFFECT_LASH_OUT:
+        if (gProtectStructs[battlerAtk].statFell == 1)
+            basePower *= 2;
+        break;
+    case EFFECT_EXPLOSION:
+        if (move == MOVE_MISTY_EXPLOSION && gFieldStatuses & STATUS_FIELD_MISTY_TERRAIN && IsBattlerGrounded(battlerAtk))
+            basePower = 150;
```
Comment (ultima-soul) [inline src/battle_util.c:6697]: This should probably be changed to a multiplication so that it's consistent with other damage boosts in this function.

**Review by ultima-soul — CHANGES_REQUESTED** [review]
Some notes from testing:
```
Surging Strikes:
    - Should be like EFFECT_DOUBLE_HIT but for 3 hits since the 3 hits are guaranteed from a single accuracy calc

Obstruct:
    - Doesn't go down in chance after successive uses
    - Sucker Punch should say "But it failed!" when used on mon that used Obstruct

Behemoth Blade/Bash:
    - Maybe renamed `EFFECT_BEHEMOTH` to `EFFECT_DYNAMAX_DOUBLE_DMG`

Decorate:
    - Should be blocked by Crafty Shield

Snipe Shot:
    - Freezes the game when used
```

**Comment by ghoulslash** [comment]
ready for re review

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
- Eerie Spell's PP drop message is bit too fast to read.
- I can't make Lash Out double its power
lashout
- Obstruct is lowering the defender's defense by 3 stages in addition to the attacker's 2 stages
obstruct
- Decorate is still not blocked by Crafty Shield.

**Comment by ghoulslash** [comment]
ready for re-review:

`Eerie Spell's PP drop message is bit too fast to read.`
- added missing `waitmessage`

`I can't make Lash Out double its power`
- merged the statFell with Eject Pack so it should work now. One was set to ProtectStruct and the other SpecialStatuses

`Obstruct is lowering the defender's defense by 3 stages in addition to the attacker's 2 stages`
- fixed, merged with Buffel's `touchedProtectLike` implementation

`Decorate is still not blocked by Crafty Shield.`
- fixed

**Comment by AsparagusEduardo** [comment]
Lash Out still doesn't want to double attack.
lashout
Everything else is ok.

File: src/battle_script_commands.c [hunk 6]
```diff
@@ -9409,7 +9409,7 @@ static u32 ChangeStatBuffs(s8 statValue, u32 statId, u32 flags, const u8 *BS_ptr
             {
                 // Check eject pack. disableEjectPack set for edge cases (e.g. attacking weak armor'd eject pack holder with u-turn)
                 if (gProtectStructs[gActiveBattler].disableEjectPack == 0)
-                    gSpecialStatuses[gActiveBattler].statFell = TRUE;
+                    gProtectStructs[gActiveBattler].statFell = 1;
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:9412]: Why the change from `TRUE` to 1?

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Review by AsparagusEduardo — APPROVED** [review]
